### PR TITLE
[18.0][MIG] delivery_carrier_pricelist: Migration to 18.0

### PR DESCRIPTION
**Pull request by chaule97**
- Since the current [delivery](https://github.com/odoo/odoo/pull/110686/files)  module depends on `sale` instead of `sale_stock`, I need to update the `delivery_carrier_pricelist` module to depend on `stock_delivery` instead of `delivery`.

### CODE REVIEW

File: delivery_carrier_pricelist/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Shipping Method Pricelist",
+    "summary": "Compute method method fees based on the product's pricelist.",
```
Comment (jbaudoux) [inline delivery_carrier_pricelist/__manifest__.py]:
```suggestion
    "summary": "Compute delivery method price based on the product's pricelist.",
```

File: delivery_carrier_pricelist/readme/DESCRIPTION.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Compute shipping methods fees based on Product Pricelists.
```
Comment (jbaudoux) [inline delivery_carrier_pricelist/readme/DESCRIPTION.md]:
```suggestion
Compute the delivery method price based on its related delivery product price according to the sales order pricelist.
```

File: delivery_carrier_pricelist/readme/DESCRIPTION.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+It allows to have different pricing per customer, prices depending on
+dates, ... The pricelist based cost is computed from the shipping
```
Comment (jbaudoux) [inline delivery_carrier_pricelist/readme/DESCRIPTION.md]:
```suggestion
dates, ... The pricelist based cost is computed from the delivery
```

File: delivery_carrier_pricelist/models/delivery_carrier.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        selection_add=[("pricelist", "Pricelist Cost")],
+        ondelete={"pricelist": "set default"},
+        help="Estimated Cost: the customer will be invoiced the estimated"
+        " cost of the shipping.\n"
+        "Real Cost: the customer will be invoiced the real cost of the"
+        " shipping, the cost of the shipping will be updated on the"
+        " SO after the delivery.\n"
+        "Pricelist Cost: the customer will be invoiced the price of the "
+        "product based on the pricelist of the sales order. The provider's "
+        "cost is ignored.",
```
Comment (jbaudoux) [inline delivery_carrier_pricelist/models/delivery_carrier.py]:
```suggestion
        selection_add=[("pricelist", "Delivery Product Price")],
        ondelete={"pricelist": "set default"},
        help="Estimated Cost: the customer will be invoiced the estimated"
        " cost of the shipping.\n"
        "Real Cost: the customer will be invoiced the real cost of the"
        " shipping, the cost of the shipping will be updated on the"
        " SO after the delivery.\n"
        "Delivery Product Price: the customer will be invoiced the price of the "
        "related delivery product based on the pricelist of the sales order. The provider's "
        "cost is ignored.",
```

File: delivery_carrier_pricelist/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Shipping Method Pricelist",
```
Comment (jbaudoux) [inline delivery_carrier_pricelist/__manifest__.py]:
```suggestion
    "name": "Delivery Method Pricelist",
```

**Review by jbaudoux — COMMENTED** [review]
Some rewording

**Review by jbaudoux — APPROVED** [review]

**Review by ajaniszewska-dev — APPROVED** [review]

**Pull request merged**